Extensions/P2P/JsExtension.js

```diff
@@ -0,0 +1,297 @@
+// @flow
+/**
+ * This is a declaration of an extension for GDevelop 5.
+ *
+ * ℹ️ Changes in this file are watched and automatically imported if the editor
+ * is running. You can also manually run `node import-GDJS-Runtime.js` (in newIDE/app/scripts).
+ *
+ * The file must be named "JsExtension.js", otherwise GDevelop won't load it.
+ * ⚠️ If you make a change and the extension is not loaded, open the developer console
+ * and search for any errors.
+ *
+ * More information on https://github.com/4ian/GDevelop/blob/master/newIDE/README-extensions.md
+ */
+
+/*::
+// Import types to allow Flow to do static type checking on this file.
+// Extensions declaration are typed using Flow (like the editor), but the files
+// for the game engine are checked with TypeScript annotations.
+import { type ObjectsRenderingService, type ObjectsEditorService } from '../JsExtensionTypes.flow.js'
+*/
+
+module.exports = {
+  createExtension: function(_/*: (string) => string */, gd/*: libGDevelop */) {
+    const extension/*: gdPlatformExtension */ = new gd.PlatformExtension();
+    extension.setExtensionInformation(
+      'P2P',
+      _('Peer-to-Peer communication (experimental)'),
+      _(
+        'Adds possibility to connect multiple game instances together via WebRTC (P2P)'
+      ),
+      'Arthur Pacaud (arthuro555)',
+      'MIT'
+    )
+    .setExtensionHelpPath('/all-features/p2p');
+
+    extension
+      .addCondition(
+        'OnEvent',
+        _('Event triggered by peer'),
+        _('Triggers once when a connected client sends the event'),
+        _('Event _PARAM0_ received from other client (data loss: _PARAM1_)'),
+        _('P2P (experimental)'),
+        'JsPlatform/Extensions/p2picon.svg',
+        'JsPlatform/Extensions/p2picon.svg'
+      )
+      .addParameter('string', _('Event name'), '', false)
+      .addParameter('yesorno', _('Data loss allowed?'), '', false)
+      .getCodeExtraInformation()
+      .setIncludeFile('Extensions/P2P/A_peer.js')
+      .addIncludeFile('Extensions/P2P/B_p2ptools.js')
+      .setFunctionName('gdjs.evtTools.p2p.onEvent');
+
+    extension
+      .addCondition(
+        'IsReady',
+        _('Is P2P ready'),
+        _('True if the peer-to-peer extension initialized and is ready to use.'),
+        _('Is P2P ready?'),
+        _('P2P (experimental)'),
+        'JsPlatform/Extensions/p2picon.svg',
+        'JsPlatform/Extensions/p2picon.svg'
+      )
+      .getCodeExtraInformation()
+      .setIncludeFile('Extensions/P2P/A_peer.js')
+      .addIncludeFile('Extensions/P2P/B_p2ptools.js')
+      .setFunctionName('gdjs.evtTools.p2p.isReady');
+
+    extension
+      .addCondition(
+        'OnError',
+        _('An error occured'),
+        _('Triggers once when an error occurs.'),
```

Comment:
Give a bit more context to the user: "Triggers once when an error occurs. Use `P2P::GetLastError()` expression to get the content of the error if you want to analyse it or display it to the user."